File: revsision/RemoveDuplicates.java
package revsision;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

public class RemoveDuplicates {

    static int unique(int[] arr) {
        Set<Integer> set = new HashSet<>();
        for (int i = 0; i < arr.length; i++) {
            set.add(arr[i]);
        }
        return set.size();
    }

    public static void main(String[] args) {
        // int j = 1;
        int[] arr = { 1, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4 };
        // for (int i = 1; i < arr.length; i++) {
        // if (arr[i] != arr[i - 1]) {
        // arr[j] = arr[i];
        // j++;
        // }
        // }
        // System.out.println(Arrays.toString(arr));
        int res = unique(arr);
        System.out.println(res);
    }
}

File: revsision/RemoveDuplicates.py
def removeDuplicates(nums):
    j=1
    for i in range(1,len(nums)):
        if(nums[i-1]!=nums[i]):
            nums[j]=nums[i]
            j+=1
    return nums
        
    
    
nums = [0,0,1,1,1,2,2,3,3,4]
removeDuplicates(nums)
print(nums)